tbot_contrib/gpio/gpio.py

```diff
@@ -0,0 +1,60 @@
+from tbot.machine import linux
+
+
+class gpio:
+    def __init__(self, lnx: linux.LinuxShell, gpio_number: str):
+        self._lnx = lnx
+        self._gpio_number = gpio_number
+        self._gpio_sysclass_path = "/sys/class/gpio/"
+        self._gpio = self._gpio_sysclass_path + "gpio" + self._gpio_number
+        self._export()
+
+    def _export(self) -> None:
+        output = self._lnx.exec0(
+            "find", self._gpio_sysclass_path, "-name", "gpio" + self._gpio_number
+        )
+        if output.strip() == (self._gpio):
+            pass
+        else:
+            self._lnx.exec0(
+                "echo",
+                self._gpio_number,
+                linux.Raw(">"),
+                self._gpio_sysclass_path + "export",
+            )
+
+    def _is_in_direction(self) -> bool:
+        output = self._lnx.exec0("cat", self._gpio + "/direction")
+        if output.strip() == "in":
+            return True
+        else:
+            return False
+
+    def set_in_direction(self) -> None:
+        if self._is_in_direction():
+            pass
+        else:
+            self._lnx.exec0("echo", "in", linux.Raw(">"), self._gpio + "/direction")
+
+    def set_out_direction(self) -> None:
+        if not self._is_in_direction():
+            pass
+        else:
+            self._lnx.exec0("echo", "out", linux.Raw(">"), self._gpio + "/direction")
+
+    def get_direction(self) -> str:
+        return self._lnx.exec0("cat", self._gpio + "/direction").strip()
+
+    def set_value(self, value: str) -> None:
+        if self._is_in_direction():
+            self.set_out_direction()
+            self._lnx.exec0("echo", value, linux.Raw(">"), self._gpio + "/value")
+        else:
+            self._lnx.exec0("echo", value, linux.Raw(">"), self._gpio + "/value")
+
+    def get_value(self) -> str:
+        if self._is_in_direction():
+            self.set_out_direction()
+            return self._lnx.exec0("cat", self._gpio + "/value").strip()
+        else:
+            return self._lnx.exec0("cat", self._gpio + "/value").strip()
```

Comment:
Not sure whether I understand what this code is doing ... Can't you just read out the value in all cases?  Why is the direction relevant here and even more, why are you switching the direction in some cases? E.g. the following should work as well?

```suggestion
        return self._lnx.exec0("cat", self._gpio + "/value").strip()
```